Registration/RegisterWithAzure.psm1

```diff
@@ -203,59 +307,84 @@ function Set-AzsRegistration{
         [Parameter(Mandatory = $true)]
         [String] $PrivilegedEndpoint,
 
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true, ParameterSetName = "Register")]
```

Comment:
It won't let me leave a comment on line 198/302, but now that we have an extra parameter set, we should set the metadata in the CmdletBinding:

[CmdletBinding(DefaultParameterSetName='Register')]